data_steward/gcloud/bq/__init__.py

```diff
@@ -227,20 +229,33 @@ def define_dataset(self, dataset_id: str, description: str,
 
         return dataset
 
-    def copy_dataset(self, input_dataset: str, output_dataset: str) -> list:
+    def copy_dataset(self,
+                     input_dataset: str,
+                     output_dataset: str,
+                     job_config: CopyJobConfig = None) -> list:
         """
         Copies tables from source dataset to a destination datasets
 
         :param input_dataset: fully qualified name of the input(source) dataset
         :param output_dataset: fully qualified name of the output(destination) dataset
+        :param job_config: An optional google.cloud.bigquery.job.CopyJobConfig
         :return: incomplete jobs
         """
+
+        if not job_config:
+            job_config = CopyJobConfig()  # create an empty default
+
         # Copy input dataset tables to backup and staging datasets
         tables = super(BigQueryClient, self).list_tables(input_dataset)
         job_list = []
         for table in tables:
             staging_table = f'{output_dataset}.{table.table_id}'
-            job = self.copy_table(table, staging_table)
+            job_config.labels = {
+                'table_name': table.table_id,
+                'copy_from': input_dataset,
+                'copy_to': output_dataset
+            }
```

Comment:
This would be better as:
```suggestion
            job_config.labels.update( {
                'table_name': table.table_id,
                'copy_from': input_dataset,
                'copy_to': output_dataset
            })
```
Above, you are allowing someone to specify a `CopyJobConfig`, but this line reassigns the `labels` dictionary, potentially wiping out any labels that the calling process has specified.  Using the `update` dictionary feature will allow the job to keep any pre-existing labels and append the extra key/value pairs.  The only catch is if the previously defined labels happen to use one of the three labels defined here.  In that case, those values would be overwritten.

Reply:
This is a great catch and definitely would introduce a bug.  I do know that @brendagutman is working on adding ownership labels.

>  The only catch is if the previously defined labels happen to use one of the three labels defined here. In that case, those values would be overwritten.

This is true too.  I'm not sure how that would happen, though.